sync/src/relayer/mod.rs

```diff
@@ -238,9 +239,9 @@ impl Relayer {
         nc: &dyn CKBProtocolContext,
         peer: PeerIndex,
         block_hash_and_number: BlockNumberAndHash,
-        mut proposals: Vec<packed::ProposalShortId>,
+        proposals: Vec<packed::ProposalShortId>,
     ) {
-        proposals.dedup();
```

Comment:
I don't think there is a duplicate here

this fn used in two way：
https://github.com/nervosnetwork/ckb/blob/develop/sync/src/relayer/block_transactions_process.rs#L104
https://github.com/nervosnetwork/ckb/blob/develop/sync/src/relayer/compact_block_process.rs#L80

Will there be no duplicate proposals in the block? @zhangsoledad

Reply:
https://github.com/nervosnetwork/ckb/blob/345a88140a9f6d41d57afa76fc9ac0bc52bdd0cf/sync/src/relayer/compact_block_process.rs#L62-L63

I couldn't find any duplicate check for `compact_block.proposals()` in the code?

Reply:
> https://github.com/nervosnetwork/ckb/blob/345a88140a9f6d41d57afa76fc9ac0bc52bdd0cf/sync/src/relayer/compact_block_process.rs#L62-L63
> 
> I couldn't find any duplicate check for `compact_block.proposals()` in the code?

CompactBlockVerifier checks whether the proposal is duplicated, and the proposal of a block is also not allowed to be duplicated, and there is a corresponding check, but we do not limit the proposal to two uncles (at most two), so it may still occur here  https://github.com/nervosnetwork/ckb/blob/develop/sync/src/relayer/block_transactions_process.rs#L104
@driftluo  help to confirm.

Reply:
Also from a defensive programming point of view, I would prefer to add processing explicitly.

Reply:
In addition, from the efficiency, it would be better to change this interface to `impl trait`
```rust
pub fn request_proposal_txs(
        &self,
        ...
        proposals: impl Iterator<Item = packed::ProposalShortId>,
    ) 
```